netbox/utilities/tests/test_forms.py

```diff
@@ -245,23 +245,39 @@ def test_set_and_range(self):
 
         self.assertEqual(sorted(expand_alphanumeric_pattern(input)), output)
 
+    # Checks for any regressions of https://github.com/netbox-community/netbox/issues/13722
+    def test_numeric_set(self):
+        input = "vlan[123,456]"
+        output = sorted([
+            'vlan123',
+            'vlan456'
+        ])
+
+        self.assertEqual(sorted(expand_alphanumeric_pattern(input)), output)
+
     def test_invalid_non_pattern(self):
-        with self.assertRaises(ValueError):
-            sorted(expand_alphanumeric_pattern('r9a'))
+        input = 'r9a'
+        output = sorted([input])
+        self.assertEqual(sorted(expand_alphanumeric_pattern(input)), output)
 
     def test_invalid_range(self):
-        with self.assertRaises(ValueError):
-            sorted(expand_alphanumeric_pattern('r[8-]a'))
+        input = 'r[8-]a'
+        output = sorted([input])
+        self.assertEqual(sorted(expand_alphanumeric_pattern(input)), output)
```

Comment:
This test doesn't reflect any change in external behaviour. Current Netbox already handles the input `r[8-]a` by just generating an interface named `r[8-]a`. The problem is that the current test is broken, and this fixes the test to reflect current reality about the behaviour before and after the change.

The problem is that current test is testing the `expand_alphanumeric_pattern` function directly, but this is subtly different to testing what happens if the users enters said string. That's because the old code that calls `expand_alphanumeric_pattern` in `ExpandableNameField.to_python` first checks if the string matches `ALPHANUMERIC_EXPANSION_PATTERN`, which the input string `r[8-]a` does not.

Therefore, if the user types in `r[8-]a` in the old code, it never even reaches `expand_alphanumeric_pattern`. But in the proposed code, it does.

So that leaves me with a choice:

1. Change the unit test to reflect the actual behaviour of the old code, thus preserving the current behaviour, considering the current behaviour to be a feature.
2. Change the code so that `r[8-]a` in fact does throw an error. This is a behaviour change or a bug fix whatever you want to call it.

In the interest of minimizing the amount of changes to behaviour, I decided to just keep the existing behaviour and make it into a feature. Although, if you will confirm to me that the *intended* behaviour is for `r[8-]a` to throw an error, I can of course patch the code and the test to reflect that.

Please let me know what you'd prefer.

Reply:
Sorry for the spam here, but as a TL;DR summary:

The current unit tests don't reflect how Netbox actually works user-facing, because the entire logic is not under test. In the interest of "not doing too much" in one commit, I've preserved the existing "bugs" and updated the unit tests to reflect those bugs as features, because it's not my place to do otherwise.